internals/overlord/tlsstate/export_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tlsstate
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"slices"
+	"time"
+)
+
+// TLSKeyPairsAll does not apply any validity filtering.
+func (m TLSManager) TLSKeyPairsAll() []*TLSKeyPair {
+	return m.tlsKeyPairs
+}
+
+func (m *TLSManager) AddNewTLSKeypair(notBefore time.Time, notAfter time.Time) (*TLSKeyPair, error) {
+	return m.addNewTLSKeyPair(notBefore, notAfter)
+}
+
+func (m TLSManager) IsTLSKeypairSupported(keypair *TLSKeyPair) bool {
+	return m.isTLSKeyPairSupported(keypair)
+}
+
+func (t TLSKeyPair) IsExpired() bool {
+	return t.isExpired()
+}
+
+// FakeSystemTime allows faking the time for the TLS manager.
+func (m *TLSManager) FakeSystemTime(date string, offset time.Duration) (restore func(), clock time.Time) {
+	layout := "2006-01-02"
+	now, err := time.Parse(layout, date)
+	if err != nil {
+		panic("invalid date string")
+	}
+	now = now.Add(offset)
+	old := systemTime
+	systemTime = func() time.Time {
+		return now
+	}
+	return func() {
+		systemTime = old
+	}, now
+}
+
+// FakeCertificateValidityPeriod allows setting how long a new certificate will be valid for.
+func (m *TLSManager) FakeCertificateValidityPeriod(valid time.Duration) (restore func()) {
+	old := certificateValidityPeriod
+	certificateValidityPeriod = valid
+	return func() {
+		certificateValidityPeriod = old
+	}
+}
+
+// AddNewUnsupportedTLSKeyPair installs an unsupported ECDSA NIST P-224
```

Comment:
I'm kind of surprised we choose to use a NIST algorithm when we could be using ed25519 instead. Is there a rationale behind this choice somewhere?

Reply:
I will change the private key / certificate algorithm to Ed25519.

```
Feature	RSA	ECDSA	Ed25519
Key Size (bits)	2048–4096	256–521	256
Signature Size (bytes)	~256–512	~64–132	64
Speed (sign/verify)	Slow/Moderate	Fast/Fast	Very Fast
TLS Support	Excellent	Very Good	Limited
Certificate Size	Large	Small	Small
Security Efficiency	Moderate	High	Very High
CA Availability	Universal	High	Low
Curve Trust Issues	N/A	Possible	None
```

Reply:
Addressed. Closing.